art/defences/detector/evasion/black_box/knn_wrapper.py

```diff
@@ -0,0 +1,70 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2021
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements wrapper for sklearn knn algorithm.
+"""
+import logging
+from typing import Callable, Union, Tuple, Any
+
+import numpy as np
+from sklearn.neighbors import NearestNeighbors
+
+from art.defences.detector.evasion.black_box.contrastive_loss import np_contrastive_loss
+from art.defences.detector.evasion.black_box.memory_queue import MemoryQueue
+
+logger = logging.getLogger(__name__)
+
+
+class NearestNeighborsWrapper:
```

Comment:
To enable different nearest neighbor searches:
- Move module to `art.estimators.neighbors.scikit.learn.py`
- Create abstract mixin-class `NeighborMixin` defining abstract method `predict` to replace the current `__call__` and `fit to replace the current `update_memory`.
- Update method names in this class